### ISSUE TITLE
Unable to deselect Species backgrounds when creating a character

### ISSUE THREAD

**Issue by macrosblackd** [CleverRaven/Cataclysm-DDA#76665]
### Describe the bug

It is currently impossible to deselect the species background when creating a new character.

Workaround is to start character creation over from scratch.

### Attach save file

N/A

### Steps to reproduce

1. Create a world with the Magiclysm mod.
2. During character creation, select one of the species backgrounds (Elf, Dwarf, Goblin, etc)
3. Try to deselect the background you chose in step 2.

### Expected behavior

The selected background should be deselected and the associated traits removed.

### Screenshots

![image](https://github.com/user-attachments/assets/fbe551ed-2112-4244-9ec8-754e8c98267a)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22631.4169 (23H2)
- Game Version: 50a97f7 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Magiclysm [magiclysm],
    Mind Over Matter [mindovermatter],
    No Hope [no_hope],
    DinoMod [DinoMod],
    Bombastic Perks [bombastic_perks],
    Martial Mastery [perk_melee_system],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Tamable Wildlife [Tamable_Wildlife]
]

**Comment by Consoleable** [comment]
Yeah i noticed this with magiclysm, there's a workaround where you can select a different starting scenario to unto it, but as is it's broken.

**Comment by Maleclypse** [comment]
Confirmed by #86814
